lib/ts/runroot.cc

```diff
@@ -129,9 +131,56 @@ runroot_handler(const char **argv, bool json)
     for (auto it : TwoPath) {
       std::string path = check_parent_path(it);
       if (!path.empty()) {
-        setenv("USING_RUNROOT", path.c_str(), true);
+        using_runroot = path;
         return;
       }
     }
   }
 }
+
+std::unordered_map<std::string, std::string>
+check_runroot()
+{
+  if (using_runroot.empty()) {
+    return std::unordered_map<std::string, std::string>{};
+  }
+
+  std::string env_path = using_runroot;
+  int len              = env_path.size();
+  if ((len + 1) > PATH_NAME_MAX) {
+    ink_fatal("TS_RUNROOT environment variable is too big: %d, max %d\n", len, PATH_NAME_MAX - 1);
+  }
+  std::ifstream file;
+  std::string yaml_path = Layout::relative_to(env_path, "runroot_path.yml");
+
+  file.open(yaml_path);
+  if (!file.good()) {
+    ink_warning("Bad env path, continue with default value");
+    return std::unordered_map<std::string, std::string>{};
+  }
+
+  std::ifstream yamlfile(yaml_path);
+  std::unordered_map<std::string, std::string> runroot_map;
+  std::string str;
+  while (std::getline(yamlfile, str)) {
+    int pos = str.find(':');
+    runroot_map[str.substr(0, pos)] = str.substr(pos + 2);
+  }
+
+  // the case for relocate (use ENV to overwrite records.config values)
+  if (runroot_map["prefix"][0] != '/') {
+    setenv("PROXY_CONFIG_CONFIG_DIR", "sysconfdir", true);
```

Comment:
This was to work around an issue with records config loads data, correct? ideally something we could fix in a different PR

Reply:
Yes, this will work around that and I will do the fix in a different PR.